File: src/main/java/ru/moscow/ucbazar/entity/Items.java
package ru.moscow.ucbazar.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity(name="items")
public class Items {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String bannerImage;


    @ElementCollection
    @CollectionTable(name = "my_entity_title", joinColumns = @JoinColumn(name = "my_entity_id"))
    @MapKeyColumn(name = "title_lang")
    @Column(name = "title_value")
    private Map<String,String> title;

    private Boolean isActive = true;


    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "Item_id")
    private List<UcEntity> uc = new ArrayList<>();


}

File: src/main/java/ru/moscow/ucbazar/model/Items.kt
package ru.moscow.ucbazar.model

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Items")
data class Items(
    @Schema(description = "banner image")
    var banner_img: String,
    @Schema(description = "title")
    var title: String,
    @Schema(description = "is Active")
    var uc: List<UcModel>
)
